conformance/lang/expressions/object-constructor-expr/object_fields.balt

```diff
@@ -0,0 +1,474 @@
+        }
+    };
+}
+
+Test-Case: error
+Description: Test whether the field is assigned to exactly once, either by its initializer or in the init method
+             when final is present.
+Labels: array-type, field-access-expr, int, list-constructor-expr, method-defn, object-constructor-expr, object-field,
+        object-type, var
+
+function errorFunction() {
+    var _ = object {
+        final int[] val;
+
+        function init() {
+            self.val = [];
+        }
+
+        function setValue() {
+            self.val = [1 ,2]; // @error cannot assign a value to final variable
+        }
+    };
+}
+
+Test-Case: output
+Description: Test object-constructor-expr constructing an object with its read-only bit set if every object-field
+             is declared as final and has a type-descriptor that is a subtype of readonly.
+Labels: any, array-type, int, intersection-type, is-expr, list-constructor-expr, method-defn, module-type-defn,
+        object-constructor-expr, object-field, object-type, readonly-type, type-cast-expr, var
+
+type ObjectType object {
+    int[] val;
+    int val2;
+};
+
+function init() {
+    var obj = object {
+        final int[] & readonly val = [];
+        final int val2 = 10;
+
+        function init() {
+        }
+    };
+    io:println(obj is readonly); // @output true
+    io:println(obj is ObjectType & readonly); // @output true
+
+    ObjectType & readonly obj2 = obj;
+    io:println(obj2.val); // @output []
+    io:println(obj2.val2); // @output 10
+
+    var obj3 = object {
+        final int[] val = [];
+        final int val2 = 10;
+
+        function init() {
+        }
+    };
+    io:println(<any>obj3 is readonly); // @output false
+    io:println(<any>obj3 is ObjectType & readonly); // @output false
+}
+
+Test-Case: error
+Description: Test object-constructor-expr constructing an object with its read-only bit set if every object-field
+             is declared as final and has a type-descriptor that is a subtype of readonly.
+Labels: array-type, int, intersection-type, list-constructor-expr, method-defn, module-type-defn,
+        object-constructor-expr, object-field, object-type, readonly-type, var
+
+type ObjectType object {
+    int[] val;
+    int val2;
+};
+
+function errorFunction() {
+    var obj = object {
+        final int[] val = [];
+        final int val2 = 10;
+
+        function init() {
+        }
+    };
+    ObjectType & readonly _ = obj; // @error incompatible types, expected '(ObjectType & readonly)'
+}
+
+Test-Case: output
+Description: Test isolatedness of object field if it is declared as final and has a type-descriptor that is a subtype of
+             readonly.
+Labels: array-type, field-access-expr, int, intersection-type, isolated-function, list-constructor-expr,
+        member-access-expr, method-call-expr, method-defn, object-constructor-expr, object-field, readonly-type, var
+
+function init() {
```

Comment:
How does this check the isolatedness of the obj field?

Reply:
The test does not properly cover this. Will update the relevant tests